File: android/app/src/main/java/my/ssdid/wallet/domain/sdjwt/SdJwtVc.kt
package my.ssdid.wallet.domain.sdjwt

data class SdJwtVc(
    val issuerJwt: String,
    val disclosures: List<Disclosure>,
    val keyBindingJwt: String?
) {
    fun present(selectedDisclosures: List<Disclosure>, kbJwt: String? = null): String {
        val parts = mutableListOf(issuerJwt)
        for (d in selectedDisclosures) {
            parts.add(d.encode())
        }
        return if (kbJwt != null) {
            parts.joinToString("~") + "~$kbJwt"
        } else if (selectedDisclosures.isNotEmpty()) {
            parts.joinToString("~") + "~"
        } else {
            issuerJwt
        }
    }
}

File: ios/SsdidWallet/Domain/SdJwt/SdJwtVc.swift
import Foundation

struct SdJwtVc {
    let issuerJwt: String
    let disclosures: [Disclosure]
    let keyBindingJwt: String?

    func present(selectedDisclosures: [Disclosure], kbJwt: String? = nil) throws -> String {
        var parts = [issuerJwt]
        for d in selectedDisclosures {
            parts.append(try d.encode())
        }
        if let kb = kbJwt {
            return parts.joined(separator: "~") + "~\(kb)"
        } else if !selectedDisclosures.isEmpty {
            return parts.joined(separator: "~") + "~"
        } else {
            return issuerJwt
        }
    }
}
